### PR TITLE
improve checks for libatomic linking

### PR DESCRIPTION
**Pull request by alexfanqi**
Some components are missing libatomic link even with #1617 on riscv. And for some reason, the atomic check introduced in commit [f502ccb](https://github.com/MariaDB/server/commit/f502ccbcb5dfce29067434885a23db8d1bd5f134) doesn't take effect (see https://bugs.gentoo.org/828065). This code piece is adapted from <https://github.com/KDE/krita/blob/451e1415fd1b3a0f9c24de2fd4707d1b1e164236/cmake/modules/CheckAtomic.cmake#L23> and seems to work better for me on riscv64.

### CODE REVIEW

File: configure.cmake [hunk 1]
```diff
@@ -874,8 +881,15 @@ ELSE()
   SET(OLD_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
   LIST(APPEND CMAKE_REQUIRED_LIBRARIES "atomic")
   CHECK_CXX_SOURCE_COMPILES("
+  #include <atomic>
+  std::atomic<int> x;
+  std::atomic<short> y;
```
Comment (robertbindar) [inline configure.cmake:886]: Does it make a difference whether we create 1 or 3 atomic variables to perform the check?

**Review by robertbindar — CHANGES_REQUESTED** [review]
Thanks for the contribution @alexfanqi! Don't have the time to look at this carefully now, @grooverdan will, but something caught my eye whilst quickly looking over the patch, I left a comment inline to make sure it is addressed before merging.

File: configure.cmake (hunk 1 above)
Comment (alexfanqi) [reply configure.cmake:886]: I am really not certain about this. It checks for different sizes of __atomic_add_fetch.  Riscv spec specifies atomic ops can be 32 bits or 64 bits (RV64 only) and doesn't support byte increment. So yeah, checking short and char should be equivalent. int is 4 bytes for both rv64 and rv32 machines and should have builtin atomics unless the machine lacks atomic extension.
But I think it is more future proof to add more checks given they aren't very expensive checks.

**Comment by alexfanqi** [comment]
no idea why the first commit fails on centos. Let me try this instead

**Comment by grooverdan** [comment]
If the centos failures where only on ARM it could be https://bugzilla.redhat.com/show_bug.cgi?id=2022588

**Comment by grooverdan** [comment]
Looks good.

As this is a bug fix for f502ccb can you rebase this on 10.3, force push to the same branch (branch name can't be changed), and change the Update branch above.

**Comment by dlan17** [comment]
Hi, I've cherry-picked the patch and apply it to 10.6.5, it works great! so to sum this: tested on Gentoo Linux, with hardware sifive unmatched box (rv64d profile)

You can have my ACK if you like: Tested-by: Yixun Lan <dlan@gentoo.org>

**Pull request merged**